syntax = "proto3";
package travel;

// ─── Enums ────────────────────────────────────────────────────────────────────
enum Trip {
  UNKNOWN_TRIP = 0;
  ROUND_TRIP   = 1;
  ONE_WAY      = 2;
  MULTI_CITY   = 3;
}

enum Passenger {
  UNKNOWN_PASSENGER = 0;
  ADULT             = 1;
  CHILD             = 2;
  INFANT_IN_SEAT    = 3;
  INFANT_ON_LAP     = 4;
}

enum Seat {
  UNKNOWN_SEAT     = 0;
  ECONOMY          = 1;
  PREMIUM_ECONOMY  = 2;
  BUSINESS         = 3;
  FIRST            = 4;
}

// ─── Leaf messages ─────────────────────────────────────────────────────────────
message Airport {
  sint32 flag    = 1;  // zig-zag(–1)=1 → bytes 08 01
  string airport = 2;
}

message ItineraryData {
  string departure_airport = 1;
  string departure_date    = 2;
  string arrival_airport   = 3;
  string flight_code       = 5;
  string flight_number     = 6;
}

message FlightData {
  string               date        = 2;
  ItineraryData        itin_data   = 4;
  optional int32                max_stops   = 5;
  repeated string      airlines    = 6;
  repeated Airport     from_flight = 13;
  repeated Airport     to_flight   = 14;
}


// ─── Top-level wrapper ─────────────────────────────────────────────────────────
message BookingRequest {
  repeated FlightData flights         = 3;  // two back-to-back FlightData blobs
  sint32              passengers_flag = 8;  // zig-zag(–1)=1 → varint 0x01
  sint32              seat_flag       = 9;  // zig-zag(–1)=1 → varint 0x01
  sint32              info_trip_flag  = 19; // zig-zag(–1)=1 → varint 0x01
}
